src/main/scala/dsptools/DspTester.scala

```diff
@@ -3,259 +3,366 @@
-            val bigIntImaginary = dspPeek(c.imaginary.asInstanceOf[DspReal]).left.get
-            Right(Complex(bigIntReal, bigIntImaginary))
-          case "SInt" =>
-            val real = peek(c.real.asInstanceOf[SInt]).toDouble
-            val imag = peek(c.imaginary.asInstanceOf[SInt]).toDouble
-            Right(Complex(real, imag))
-          case _ =>
-            throw DspException(
-              s"peek($c): c DspComplex has unknown underlying type ${c.getClass.getName}")
+  def poke(c: DspComplex[_], value: Complex): Unit = {
+    updatableDspVerbose.withValue(dispBits) {
+      (c.real, c.imag) match {
+        case (real: Data, imag: Data) => {
+          poke(real, value.real)
+          poke(imag, value.imag)
         }
-      case r: DspReal =>
-        val bigInt = peek(r.node)
-        Left(bigIntBitsToDouble(bigInt))
-      case r: FixedPoint =>
-        val bigInt = peek(r.asInstanceOf[Bits])
-        Left(FixedPoint.toDouble(bigInt, r.binaryPoint.get))
-      case s: SInt =>
-        Left(peek(s).toDouble)
-      case _ =>
-        throw DspException(s"peek($data): data has unknown type ${data.getClass.getName}")
+      }
     }
+    if (dispDsp) logger println s"  POKE ${getName(c)} <- ${value.toString}"
   }
 
-  // [stevo]: peek a UInt but cast it to another type
-  def dspPeekAs[U<:Data](data: Data, typ: U): Either[Double, Complex] = {
-    data match {
-      case u: UInt =>
-        typ match {
-          // TODO:
-          //case c: DspComplex[_] =>
-          //  c.underlyingType() match {
-          //    case "fixed" =>
-          //      val real      = dspPeek(c.real.asInstanceOf[FixedPoint]).left.get
-          //      val imaginary = dspPeek(c.imaginary.asInstanceOf[FixedPoint]).left.get
-          //      Right(Complex(real, imaginary))
-          //    case "real"  =>
-          //      val bigIntReal      = dspPeek(c.real.asInstanceOf[DspReal]).left.get
-          //      val bigIntImaginary = dspPeek(c.imaginary.asInstanceOf[DspReal]).left.get
-          //      Right(Complex(bigIntReal, bigIntImaginary))
-          //    case "SInt" =>
-          //      val real = peek(c.real.asInstanceOf[SInt]).toDouble
-          //      val imag = peek(c.imaginary.asInstanceOf[SInt]).toDouble
-          //      Right(Complex(real, imag))
-          //    case _ =>
-          //      throw DspException(
-          //        s"peek($c): c DspComplex has unknown underlying type ${c.getClass.getName}")
-          //  }
-          case _: DspReal =>
-            val bigInt = peek(u)
-            Left(bigIntBitsToDouble(bigInt))
-          case r: FixedPoint =>
-            val bigInt = peek(u)
-            Left(toDoubleFromUnsigned(bigInt, r.getWidth, r.binaryPoint.get))
-          // TODO:
-          //case s: SInt =>
-          //  Left(peek(s).toDouble)
-          case _ =>
-            throw DspException(s"peek($data): data has unknown type ${data.getClass.getName}")
```

Comment:
I think a peek as UInt should exist elsewhere... (esp. if it's singularly for testing your DspBlock stuff, then it should go in your DspBlockTester) but you can debate with me on why you might not want that.